client/packages/common/src/authentication/api/hooks/useRefreshToken.ts

```diff
@@ -40,6 +40,7 @@ export const useRefreshToken = () => {
         setHeader('Authorization', `Bearer ${token}`);
       });
     }
+    if (minutesSinceLastRequest >= COOKIE_LIFETIME_MINUTES) onTimeout();
```

Comment:
If you make this an early return and move it before the refresh api call:

```suggestion
    if (minutesSinceLastRequest >= COOKIE_LIFETIME_MINUTES) {
      onTimeout();
      return;
    }
```

then you can remove the `minutesSinceLastRequest` check, rather than repeating it

```
    const expiresSoon = expiresIn === 1 || expiresIn === 2;
    if (expiresSoon) {
      mutateAsync().then(data => {
        const token = data?.token ?? '';
        const newCookie = { ...authCookie, token };
        setAuthCookie(newCookie);
        setHeader('Authorization', `Bearer ${token}`);
      });
    }
```